### DISCUSSION TITLE
MS2 filter out in the feature table?

### DISCUSSION THREAD

**Discussion by marwa38** [mzmine/mzmine#2097] [General]
Hi team 

I am using mzmine v4.2 on windows.

I ran the mzmine already and exported as MS2 or ion identity as in the screenshot and I don't want to run the analysis again from start as my datasets are quite big and time consuming. 
Anyway I could filter only by MS2 in my features.csv table? want to have ms2 only. 
<img width="438" alt="image" src="https://github.com/user-attachments/assets/b8432f41-2238-4903-8125-615a08317678">
These are the column header names below (auto MS2 verify doesn't have anything) but I can't find something that could filter MS2 out, ANY HELP PLEASE?
row ID
row m/z
row retention time
row ion mobility
row ion mobility unit
row CCS
correlation group ID
annotation network number
best ion
auto MS2 verify
identified by n=
partners
neutral M mass

Thanks 
Marwa

**Comment by robinschmid** [comment]
If you want to filter your output CSV you can load both the CSV and the gnps mgf file in python. The row ID is in both the mgf and the CSV so you can filter it.

**Comment by marwa38** [comment]
i don't see any column in the .csv that distinguish between MS1 and 2 to filter them out that is why I am asking

**Comment by marwa38** [comment]
I checked the .mgf file and found no MSLEVEL=1
Weird shouldn't it be MS1? as they are exported as "MS2 or ion identity"
![image](https://github.com/user-attachments/assets/e5752f75-270c-406f-84ca-fda7b2312f11)

**Comment by robinschmid** [comment]
All the features in csv are exported for their MS1 height or area. A filter is applied to only retain features that have MS2 or an ion identity. The molecular networking mgf contains a single MS2 spectrum per feature. So features that have no MS2 and are only exported for their ion identity will be present in the CSV but not in the mgf.
